eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -197,39 +197,97 @@ object Transactions {
     } else {
       (spec.toLocal.truncateToSatoshi, spec.toRemote.truncateToSatoshi - commitFee)
     } // NB: we don't care if values are < 0, they will be trimmed if they are < dust limit anyway
-
-    val toLocalDelayedOutput_opt = if (toLocalAmount >= localDustLimit) Some(TxOut(toLocalAmount, pay2wsh(toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey)))) else None
-    val toRemoteOutput_opt = if (toRemoteAmount >= localDustLimit) Some(TxOut(toRemoteAmount, pay2wpkh(remotePaymentPubkey))) else None
-
-    val htlcOfferedOutputs = trimOfferedHtlcs(localDustLimit, spec)
-      .map(htlc => TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)))))
-    val htlcReceivedOutputs = trimReceivedHtlcs(localDustLimit, spec)
-      .map(htlc => TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcReceived(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes), htlc.add.cltvExpiry))))
-
+    val outputs = collection.mutable.ArrayBuffer.empty[CommitmentOutputLink]
+
+    if (toLocalAmount >= localDustLimit) outputs.append(
+      CommitmentOutputLink(
+        TxOut(toLocalAmount, pay2wsh(toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey))),
+        toLocalDelayed(localRevocationPubkey, toLocalDelay, localDelayedPaymentPubkey),
+        ToLocalLink))
+
+    if (toRemoteAmount >= localDustLimit) outputs.append(
+      CommitmentOutputLink(
+        TxOut(toRemoteAmount, pay2wpkh(remotePaymentPubkey)),
+        pay2pkh(remotePaymentPubkey),
+        ToRemoteLink))
+
+    trimOfferedHtlcs(localDustLimit, spec).foreach(htlc =>
+      outputs.append(CommitmentOutputLink(
+        TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)))),
+        htlcOffered(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes)),
+        htlc)))
+
+    trimReceivedHtlcs(localDustLimit, spec).foreach(htlc =>
+      outputs.append(CommitmentOutputLink(
+        TxOut(htlc.add.amountMsat.truncateToSatoshi, pay2wsh(htlcReceived(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes), htlc.add.cltvExpiry))),
+        htlcReceived(localHtlcPubkey, remoteHtlcPubkey, localRevocationPubkey, ripemd160(htlc.add.paymentHash.bytes), htlc.add.cltvExpiry),
+        htlc)))
+
+    outputs.sortWith(CommitmentOutputLink.sort).toIndexedSeq
+  }
+
+  def makeCommitTx(commitTxInput: InputInfo,
+                   commitTxNumber: Long,
+                   localPaymentBasePoint: PublicKey,
+                   remotePaymentBasePoint: PublicKey,
+                   localIsFunder: Boolean,
+                   outputs: IndexedSeq[CommitmentOutputLink]): CommitTx = {
```

Comment:
nit: is it important this it's an `IndexedSeq` and not just a `Seq`? Maybe I'm missing it, but it seems we're not using random-access to the elements anywhere, we're always zipping with index.

Reply:
It isn't, I've made is a simple `Seq` in the latest commit